### PR TITLE
Add CLI basics in docs

### PR DESCRIPTION
**Pull request by GantMan**
Adding documentation for the CLI for 2 reasons:

1.  It's not immediately clear to most how the CLI is part of `react-native` and not the node module they installed.  This begins clarifying.
2. I plan on adding some additional documentation to this section pending an upcoming PR I've discussed with Mike.

screen shot provided:
image

### CODE REVIEW

File: docs/UnderstandingCLI.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+id: understanding-cli
+title: Understanding CLI
```
Comment (mkonicek) [inline docs/UnderstandingCLI.md]: nit: Understanding the CLI

File: docs/UnderstandingCLI.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Though you may have installed the `react-native-cli` via npm as a separate module, it is a shell for accessing the CLI embedded
+in the React Native of each project. Your commands and their effects are dependent on the version of the module of `react-native` 
+in context of the project. This guide will give a brief overview of the CLI in the module.
+
+# Location and import of the local CLI
+
+React Native has a [`local-cli`](https://github.com/facebook/react-native/tree/master/local-cli) folder with a file named 
+[`cliEntry.js`](https://github.com/facebook/react-native/blob/master/local-cli/cliEntry.js).  Here, the commands are read 
+from `commands.js` and added as possible CLI commands.  _E.G._ the `react-native link` command, existis in the
```
Comment (mkonicek) [inline docs/UnderstandingCLI.md]: nit: exists

File: docs/UnderstandingCLI.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Though you may have installed the `react-native-cli` via npm as a separate module, it is a shell for accessing the CLI embedded
+in the React Native of each project. Your commands and their effects are dependent on the version of the module of `react-native` 
+in context of the project. This guide will give a brief overview of the CLI in the module.
+
+# Location and import of the local CLI
+
+React Native has a [`local-cli`](https://github.com/facebook/react-native/tree/master/local-cli) folder with a file named 
+[`cliEntry.js`](https://github.com/facebook/react-native/blob/master/local-cli/cliEntry.js).  Here, the commands are read 
+from `commands.js` and added as possible CLI commands.  _E.G._ the `react-native link` command, existis in the 
+[`react-native/local-cli/link`](https://github.com/facebook/react-native/blob/master/local-cli/link/link.js) folder, and is
```
Comment (mkonicek) [inline docs/UnderstandingCLI.md]: nit: says folder but links to a file

File: docs/UnderstandingCLI.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+imported/required in `commands.js`, which will register it as a documented command to be exposed to the CLI.
+
+# Command protocol
+
+At the end of each command entry is an export.  The export is an object with a function to perform, description of the command, and the command name.  The object structure for the `unlink` command looks like so:
```
Comment (mkonicek) [inline docs/UnderstandingCLI.md]: How about using `link` here since it's used as an example above, to keep it simple and not to confuse people?

**Comment by mkonicek** [comment]
This is a great idea, thanks for doing it!

File: docs/UnderstandingCLI.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+# Location and import of the local CLI
+
+React Native has a [`local-cli`](https://github.com/facebook/react-native/tree/master/local-cli) folder with a file named 
+[`cliEntry.js`](https://github.com/facebook/react-native/blob/master/local-cli/cliEntry.js).  Here, the commands are read 
+from `commands.js` and added as possible CLI commands.  _E.G._ the `react-native link` command, existis in the 
+[`react-native/local-cli/link`](https://github.com/facebook/react-native/blob/master/local-cli/link/link.js) folder, and is
+imported/required in `commands.js`, which will register it as a documented command to be exposed to the CLI.
+
+# Command protocol
```
Comment (mkonicek) [inline docs/UnderstandingCLI.md]: How about 'Command definitions'? It's not really a protocol in the sense of a communication protocol.

**Comment by GantMan** [comment]
@mkonicek - nits added.  Great feedback.

**Comment by mkonicek** [comment]
Looks good, thanks!

@facebook-github-bot shipit

**Pull request closed without merging**